api/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.local_ratelimit.v3;
+
+import "envoy/config/core/v3/base.proto";
+import "envoy/type/v3/http_status.proto";
+import "envoy/type/v3/token_bucket.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.local_ratelimit.v3";
+option java_outer_classname = "LocalRateLimitProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Local Rate limit]
+// Local Rate limit :ref:`configuration overview <config_http_filters_local_rate_limit>`.
+// [#extension: envoy.filters.http.local_ratelimit]
+
+// [#next-free-field: 7]
+message LocalRateLimit {
+  // The human readable prefix to use when emitting stats.
+  string stat_prefix = 1;
+
+  // This field allows to send a HTTP response status code to the downstream client other
+  // than 429 (TooManyRequests) when the request has been rate limited.
+  type.v3.HttpStatus status = 2;
+
+  // The token bucket configuration to use for rate limiting requests that are processed by this
+  // filter. Each request processed by the filter consumes a single token. If the token is available,
+  // the request will be allowed. If no tokens are available, the request will receive the configured
+  // rate limit status.
+  //
+  // .. note::
+  //   It's fine for the token bucket to not be set for the global configuration, but it must be set
+  //   for the per route configuration.
```

Comment:
I assume if it's not set on either this will result in an error? Though, this would be a runtime error? Can you clarify how this is handled if it's not set in both places? I assume no rate limiting?

Reply:
The common pattern would be:

* filter installed on the listener but doesn't apply limits (because the `filter_enabled` runtime fractional defaults to false)
* per route configs are installed via `typed_per_filter_config` (with `filter_enabled` set to true)

So I think we should throw an error in `createRouteSpecificFilterConfigTyped()` if the token bucket field is missing, like the `header_to_metadata` filter does:

https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc#L102 

I'll also clarify this a bit more in the proto file _and_ in the docs.